File: lib/antlr-2.7.5/examples/csharp/heteroAST/MULTNode.cs
using System;
using System.IO;
using antlr;
using antlr.collections;

/** A simple node to represent MULT operation */
public class MULTNode : BinaryOperatorAST {
	public MULTNode() {
	}

	public MULTNode(Token tok) {
	}

	/** Compute value of subtree; this is heterogeneous part :) */
	public override int Value() {
		return Left().Value() * Right().Value();
	}

	public override string ToString() {
		return " *";
	}

	public override void xmlSerializeRootOpen(TextWriter outWriter) {
		outWriter.Write("<MULT>");
	}

	public override void xmlSerializeRootClose(TextWriter outWriter) {
		outWriter.Write("</MULT>");
	}
}

File: lib/antlr-2.7.5/examples/java/heteroAST/MULTNode.java
import antlr.BaseAST;
import antlr.Token;
import antlr.collections.AST;
import java.io.*;

/** A simple node to represent MULT operation */
public class MULTNode extends BinaryOperatorAST {
	public MULTNode(Token tok) {
	}

	/** Compute value of subtree; this is heterogeneous part :) */
	public int value() {
		return left().value() * right().value();
	}

	public String toString() {
		return " *";
	}

	public void xmlSerializeRootOpen(Writer out) throws IOException {
		out.write("<MULT>");
	}

	public void xmlSerializeRootClose(Writer out) throws IOException {
		out.write("</MULT>");
	}
}
